app/src/main/java/com/practicum/playlistmaker/presentation/presenter/player/PlayerPresenter.kt

```diff
@@ -0,0 +1,118 @@
+package com.practicum.playlistmaker.presentation.presenter.player
+
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import com.practicum.playlistmaker.creator.Creator
+import com.practicum.playlistmaker.domain.consumer.Consumer
+import com.practicum.playlistmaker.domain.consumer.ConsumerData
+import com.practicum.playlistmaker.domain.model.PlayerState
+import com.practicum.playlistmaker.domain.model.Track
+import com.practicum.playlistmaker.presentation.mapper.TrackMapper
+
+class PlayerPresenter(
+    private val view: PlayerView,
+    private val context: Context,
+) {
+
+    private val playerInteractor = Creator.providePlayerInteractor(context)
+    private var playerState : PlayerState  = PlayerState.STATE_DEFAULT
+
+    private val handler = Handler(Looper.getMainLooper())
+    private var currentConsumeRunnable: Runnable? = null
+
+    fun loadTrack(){
+        playerInteractor.getTrackDetails(
+            consumer = object : Consumer<Track>{
+                override fun consume(data: ConsumerData<Track>) {
+                    handler.removeCallbacksSafe(currentConsumeRunnable)
+
+                    val consumeRunnable = getConsumeRunnable(data)
+                    currentConsumeRunnable = consumeRunnable
+
+                    handler.post(consumeRunnable)
+                }
+
+            }
+        )
+    }
+
+    private fun getConsumeRunnable(data: ConsumerData<Track>): Runnable {
+        return Runnable {
+            when (data) {
+                is ConsumerData.Error -> {
```

Comment:
```suggestion
                is ConsumerData.Error ->  Unit
```
Или все-таки прописать обработку ошибки